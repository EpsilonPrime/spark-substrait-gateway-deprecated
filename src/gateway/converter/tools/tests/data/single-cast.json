{
  "extensionUris": [
    {
      "extensionUriAnchor": 1,
      "uri": "/functions_boolean.yaml"
    }
  ],
  "extensions": [
    {
      "extensionFunction": {
        "extensionUriReference": 1,
        "name": "or:bool"
      }
    }
  ],
  "relations": [
    {
      "root": {
        "input": {
          "project": {
            "input": {
              "read": {
                "common": {
                  "direct": {}
                },
                "baseSchema": {
                  "names": [
                    "r_regionkey",
                    "r_name",
                    "r_comment"
                  ],
                  "struct": {
                    "types": [
                      {
                        "i32": {
                          "nullability": "NULLABILITY_REQUIRED"
                        }
                      },
                      {
                        "string": {
                          "nullability": "NULLABILITY_REQUIRED"
                        }
                      },
                      {
                        "string": {
                          "nullability": "NULLABILITY_NULLABLE"
                        }
                      }
                    ],
                    "nullability": "NULLABILITY_REQUIRED"
                  }
                },
                "namedTable": {
                  "names": [
                    "#2"
                  ]
                }
              }
            },
            "expressions": [
              {
                "cast": {
                  "type": {
                    "fixedChar": {
                      "length": 25,
                      "nullability": "NULLABILITY_REQUIRED"
                    }
                  },
                  "input": {
                    "literal": {
                      "fixedChar": "DELIVER IN PERSON"
                    }
                  }
                }
              }
            ]
          }
        }
      }
    }
  ]
}