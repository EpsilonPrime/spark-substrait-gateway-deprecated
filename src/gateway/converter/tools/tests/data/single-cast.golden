{
  "relations": [
    {
      "root": {
        "input": {
          "project": {
            "common": {
              "output_mapping": [
                0,
                1,
                2
              ]
            },
            "input": {
              "project": {
                "common": {},
                "input": {
                  "read": {
                    "common": {
                      "direct": {}
                    },
                    "baseSchema": {
                      "names": [
                        "r_regionkey",
                        "r_name",
                        "r_comment"
                      ],
                      "struct": {
                        "types": [
                          {
                            "i32": {
                              "nullability": "NULLABILITY_REQUIRED"
                            }
                          },
                          {
                            "string": {
                              "nullability": "NULLABILITY_REQUIRED"
                            }
                          },
                          {
                            "string": {
                              "nullability": "NULLABILITY_NULLABLE"
                            }
                          }
                        ],
                        "nullability": "NULLABILITY_REQUIRED"
                      }
                    },
                    "namedTable": {
                      "names": [
                        "#2"
                      ]
                    }
                  }
                },
                "expressions": [
                  {
                    "cast": {
                      "type": {
                        "fixedChar": {
                          "length": 25,
                          "nullability": "NULLABILITY_REQUIRED"
                        }
                      },
                      "input": {
                        "selection": {
                          "directReference": {
                            "structField": {
                              "field": 3
                            }
                          },
                          "rootReference": {}
                        }
                      }
                    }
                  }
                ]
              }
            },
            "expressions": [
              {
                "literal": {
                  "fixedChar": "DELIVER IN PERSON"
                }
              }
            ]
          }
        }
      }
    }
  ]
}